iothub_client/src/iothubtransport_amqp_device.c

```diff
@@ -854,52 +912,22 @@ int amqp_device_stop(AMQP_DEVICE_HANDLE handle)
         AMQP_DEVICE_INSTANCE* instance = (AMQP_DEVICE_INSTANCE*)handle;
 
         // Codes_SRS_DEVICE_09_025: [If the device state is already DEVICE_STATE_STOPPED or DEVICE_STATE_STOPPING, amqp_device_stop shall return a non-zero result]
-        if (instance->state == DEVICE_STATE_STOPPED || instance->state == DEVICE_STATE_STOPPING)
+        if (instance->state == DEVICE_STATE_STOPPED || instance->state == DEVICE_STATE_STOPPING && instance->stop_delay_ms == 0)
```

Comment:
Compile error

/Users/runner/work/1/s/iothub_client/src/iothubtransport_amqp_device.c:915:97: error: '&&' within '||' [-Werror,-Wlogical-op-parentheses]
        if (instance->state == DEVICE_STATE_STOPPED || instance->state == DEVICE_STATE_STOPPING && instance->stop_delay_ms == 0)
                                                    ~~ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/Users/runner/work/1/s/iothub_client/src/iothubtransport_amqp_device.c:915:97: note: place parentheses around the '&&' expression to silence this warning
        if (instance->state == DEVICE_STATE_STOPPED || instance->state == DEVICE_STATE_STOPPING && instance->stop_delay_ms == 0)
                                                                                                ^
                                                       (                                                                       )

Reply:
got it, thanks. Fixed.